### PR TITLE
Tag temp files created by the shared cache

### PR DESCRIPTION
**Pull request by JakeSiFive**
If the daemon is killed by an external agent at *just* the right time, there's a about a 1/1000 chance that it will happen between the creation of a temp file and the rename. This will leave a temp file sitting around. This on its own is not really a problem. The temp file can be confusing to debug the existence however and if you have tools in place to track that cleaning your workspace actually cleans everything, this will trigger that check.

A full solution would be to have the client copy out of the cache instead of having the daemon do the copy. But even that might leave temp files sitting around that you might not want. To combat this we need to tag these files so that such a tool can identify them. Later we will move the copy to the client where the odds of this sort of thing occurring drastically shrink and the odds of a user caring for that particular failure mode are even lower.

### CODE REVIEW

**Review by V-FEXrt — APPROVED** [review]

File: src/job_cache/daemon_cache.cpp [hunk 1]
```diff
@@ -906,7 +906,7 @@ FindJobResponse DaemonCache::read(const FindJobRequest &find_request) {
 
       // Finally copy the file (as efficently as we can) to
       // the destination.
-      std::string tmp_dst = pair.first + "." + rng.unique_name();
+      std::string tmp_dst = pair.first ".local-cache-tmp." + rng.unique_name();
```
Comment (V-FEXrt) [inline src/job_cache/daemon_cache.cpp:909]: hold up, this needs a `+` right?

Comment (JakeSiFive) [reply src/job_cache/daemon_cache.cpp]: yeah lol, I just fixed it

**Review by V-FEXrt — CHANGES_REQUESTED** [review]
blocking for a comment

**Comment by JakeSiFive** [comment]
> blocking for a comment

I already fixed it 😭

**Review by V-FEXrt — APPROVED** [review]
speedy

**Pull request merged**